test/types/object-type.spec.js

```diff
@@ -38,6 +38,73 @@ describe('object type', function () {
         });
     });
 
+    describe('#equal', function () {
```

Comment:
Why the hash character?

Reply:
It's usually shorthand for `.prototype.`, so it doesn't make sense here. But we should also remove it from `describe('#diff', function () {` above.

Reply:
I see. Let's keep it the as what we do in other places then.

Reply:
Yeah, I was just going by the 'diff' case - will have a look around the tree to see what's most common and then align all of them that way (since this branch adds quite a few checks).